lmdeploy/pytorch/kernels/cuda/apply_rotary_pos_emb.py

```diff
@@ -83,44 +82,44 @@ def apply_rotary_pos_emb_qk_kernel(
     sin_l = tl.load(SIN + cs_offset_l).to(q_elem_type)
     sin_h = tl.load(SIN + cs_offset_h).to(q_elem_type)
 
-    q_ptr = Q + pos_offset * stride_qs
-    qe_ptr = Q_EMB + pos_offset * stride_qes
-    ql_ptrs = q_ptr[:, None] + feat_offset_l[None, :] * stride_qd
-    qh_ptrs = q_ptr[:, None] + feat_offset_h[None, :] * stride_qd
-    qel_ptrs = qe_ptr[:, None] + feat_offset_l[None, :] * stride_qed
-    qeh_ptrs = qe_ptr[:, None] + feat_offset_h[None, :] * stride_qed
-    for _ in range(BLOCK_QH):
+    if head_id < BLOCK_QH:
+        q_ptr = Q + pos_offset * stride_qs
+        qe_ptr = Q_EMB + pos_offset * stride_qes
+        ql_ptrs = q_ptr[:, None] + feat_offset_l[None, :] * stride_qd
+        qh_ptrs = q_ptr[:, None] + feat_offset_h[None, :] * stride_qd
+        qel_ptrs = qe_ptr[:, None] + feat_offset_l[None, :] * stride_qed
+        qeh_ptrs = qe_ptr[:, None] + feat_offset_h[None, :] * stride_qed
+        ql_ptrs += head_id * stride_qh
+        qh_ptrs += head_id * stride_qh
+        qel_ptrs += head_id * stride_qeh
```

Comment:
Is it possible that `stride_qeh` is not equal to `stride_qh`?

Reply:
q k can be slice of qkv tensor. stride can be different if output is not inplaced.